LegendaryExplorer/LegendaryExplorer/Tools/PackageEditor/Experiments/PackageEditorExperimentsSquid.cs

```diff
@@ -1,28 +1,1037 @@
+                    var currentBone = psk.Bones[i];
+                    meshBin.NameIndexMap.Add(currentBone.Name, i);
+                    meshBin.RefSkeleton[i] = new MeshBone()
+                    {
+                        Name = currentBone.Name,
+                        NumChildren = currentBone.NumChildren,
+                        BoneColor = new LegendaryExplorerCore.SharpDX.Color(new Vector4(1, 1, 1, 1)),
+                        Flags = currentBone.Flags,
+                        ParentIndex = currentBone.ParentIndex,
+                        Position = new Vector3(currentBone.Position.X, currentBone.Position.Y * -1, currentBone.Position.Z),
+                        Orientation = new Quaternion(currentBone.Rotation.X, currentBone.Rotation.Y * -1, currentBone.Rotation.Z, currentBone.Rotation.W)
+                    };
+
+                    // make sure we calculate the depth
+                    GetDepth(i);
+                }
+
+                // now find the maximum depth and set that as the skeletal depth
+                meshBin.SkeletalDepth = skeletalDepth.Max();
+            }
+
+            static void SetupBounds(PSK psk, SkeletalMesh meshBin)
+            {
+                // bounds are important at least for the camera display preview in LEX, and possibly important for when to cull meshes based on visibility in game
+                // separate out the coordinates for each axis so we can operate on them
+                var xCoords = psk.Points.Select(x => x.X);
+                var yCoords = psk.Points.Select(x => -x.Y);
+                var zCoords = psk.Points.Select(x => x.Z);
+
+                // get the origin by averaging all vertex positions; it'll probably be close enough
+                var origin = new Vector3(xCoords.Average(), yCoords.Average(), zCoords.Average());
+
+                var xRange = xCoords.Select(coord => Math.Abs(coord - origin.X)).Max();
+                var yRange = yCoords.Select(coord => Math.Abs(coord - origin.Y)).Max();
+                var zRange = zCoords.Select(coord => Math.Abs(coord - origin.Z)).Max();
+                var boxExtent = new Vector3(xRange, yRange, zRange);
+
+                var sphereRad = boxExtent.Length();
+                meshBin.Bounds = new BoxSphereBounds
+                {
+                    Origin = origin,
+                    // best guess at a reasonable margin
+                    BoxExtent = boxExtent * 2,
+                    SphereRadius = sphereRad * 2
+                };
+            }
+
+            static void SetupMaterials(PackageEditorWindow pew, PSK psk, SkeletalMesh meshBin)
+            {
+                SetNumMaterialSlots(meshBin, psk.Materials.Count);
+                for (int i = 0; i < psk.Materials.Count; i++)
+                {
+                    // Does not work because it is looking for the full instanced path; can I export using that?
+                    var entry = pew.Pcc.FindEntry(psk.Materials[i].Name);
```

Comment:
If you can, you should also qualify this with the type, as some objects in vanilla game have duplicate names (looking you at you ME1).